snuba/migrations/snuba_migrations/system/0001_migrations.py

```diff
@@ -0,0 +1,28 @@
+from typing import Sequence
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration
+from snuba.migrations import operations
+
+
+class Migration(migration.Migration):
```

Comment:
So now do we depend on the number of the migration for the order or on the dependency defined by a migration?
What if we just registered them in the group class? Is there any gain in just loading all the files of a module ? If we had an explicit list it would be very hard to accidentally commit two conflicting migrations with the same number.

Reply:
No, the number of the migration never matters (nor does the ID of the migration have to be a number at all). Each migration has a single dependency property which refers to the previous migration. It's definitely possible to register them in the group instead. My idea around doing it this way instead was to make it standard across all migrations regardless of the type of group loader used and make it possible to later autogenerate migration files.

Reply:
Though how are we dealing with two people accidentally committing at the same time two migrations that depend on the same ? If each migration is the sole class with the authority of saying which migration it depends onto, this situation is possible and we should not mutate migrations to fix it. 
We need to make that impossible, possibly structurally (provide the list of migrations in order somehow, either in a module or outside if you plan to autogenerate migrations) or, less preferably, at test time by ensuring that a commit would not pass tests if this condition is violated, this is still not great because it is not impossible to commit something with failing tests.

Reply:
My plan is to do it at test time or anytime migrate script is run. Any build or test would fail. Since the snuba repository requires that builds are up to date with master before being merged, I'm not sure this is really likely, nor would mutating a migration to point at a different one be a huge deal if this somehow happened and those rules were overridden.

The alternative is to give the GroupLoader the sole responsibility of defining the order. The group loader maintains a list of IDs which matches the file names of the migrations. There is no such dependency field in the migration. I'm not opposed to switching to this approach.

Reply:
Latest commit moves the responsibility for providing the order into the group loader and removes the dependency field from the migration class. I'd leave the rest of the implementation around how these dependencies are actually managed for a follow up though.

Reply:
ok.